quickwit-metastore/src/metastore/delete_task.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use quickwit_proto::metastore_api::{DeleteQueryRequest, DeleteTaskResponse};
+use serde::{Deserialize, Serialize};
+
+/// A delete query.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct DeleteQuery {
+    /// Index ID.
+    pub index_id: String,
```

Comment:
If the types are really the same (this is the case here), you are better off using the protobuf directly.

Reply:
I was hesitating on this actually. Protobuf is always putting an option so it's a bit annoying but the extra code for just removing that option is too big :/

Reply:
I switched to protobuf.